app/models/spree/calculator/shipping/store_distance.rb

```diff
@@ -0,0 +1,89 @@
+require 'pry'
+require_dependency 'spree/shipping_calculator'
+
+module Spree
+  module Calculator::Shipping
+    class StoreDistance < Spree::ShippingCalculator
+      preference :base_amount_up_to_2_5_mi, :decimal, default: 2.5
+      preference :base_amount_up_to_3_5_mi, :decimal, default: 3.5
+      preference :base_amount_up_to_6_mi, :decimal, default: 4.5
+      preference :base_amount_up_to_9_mi, :decimal, default: 5.5
+
+      def self.description
+        Spree.t(:shipping_store_distance)
+      end
+
+      def compute_package(package)
+        compute_store_distance(package)
+      end
+
+      def vendors_coordinates(package)
+        package.order.variants.map do |variant|
+          variant.vendor.to_coordinates
+        end
+      end
+
+      def total_product_weight(package)
+        total_weight = 0
+
+        total_weight = package.order.variants.map do |variant|
+          variant.weight
+        end
+        total_weight.sum
+      end
+
+      def total_distance(package)
+        vendor_co = vendors_coordinates(package)
+
+        total_distance = 0
+
+        total_distance += Geocoder::Calculations.distance_between(package.stock_location.to_coordinates, vendor_co[0])
+
+        vendor_co[0...-1].each_with_index do |val, index|
+          total_distance += Geocoder::Calculations.distance_between(vendor_co[index], vendor_co[index + 1])
+        end
+
+        total_distance += Geocoder::Calculations.distance_between(vendor_co.last, package.order.ship_address.to_coordinates)
+
+        total_distance += Geocoder::Calculations.distance_between(package.order.ship_address.to_coordinates, package.stock_location.to_coordinates)
+      end
+
+      def store_collection_charge(package)
+        vendor_co = vendors_coordinates(package)
+        vendor_co.inject(-0.5) { |sum, number| sum + 0.5}
```

Comment:
could be a one-liner `vendors_coordinates(package).inject(-0.5) { |sum, number| sum + 0.5}`